File: BaekJoon/Stack/10828.py
import sys


N = int(sys.stdin.readline())
stack = []
top = 0
for _ in range(N):
    command = sys.stdin.readline().rstrip().split()
    if command[0] == "push":
        stack.append(int(command[1]))
        top += 1
    elif command[0] == "pop":
        if top == 0:
            print(-1)
        else:
            print(stack.pop())
            top -= 1
    elif command[0] == "size":
        print(top)
    elif command[0] == "empty":
        if top == 0:
            print(1)
        else :
            print(0)
    elif command[0] == "top":
        if top == 0 :
            print(-1)
        else:
            print(stack[-1])

File: BaekJoonSwift/BaekJoonSwift/Stack (1)/10828.swift
import Foundation

var stack = [Int]()

let n = Int(readLine()!)!

for _ in 0..<n {
    let temp = readLine()!.split(separator: " ")
    if temp[0] == "push" {
        stack.append(Int(temp[1])!)
    }
    else if temp[0] == "size" {
        print(stack.count)
    }
    else if temp[0] == "top" {
        if stack.count == 0 {
            print(-1)
        }
        else {
            print(stack.last!)
        }
    }
    else if temp[0] == "pop" {
        if stack.count == 0 {
            print(-1)
        }
        else {
            print(stack.last!)
            stack.removeLast()
        }
    }
    else if temp[0] == "empty" {
        if stack.count == 0 {
            print(1)
        }
        else {
            print(0)
        }
    }
}
